// ecoji-js/dist/src/Mapping.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const padding = 0x2615;
const padding40 = 0x269C;
const padding41 = 0x1F3CD;
const padding42 = 0x1F4D1;
const padding43 = 0x1F64B;
class Mapping {
    constructor(emojis) {
        // This should sort before everything.  This is output when 3 or less input bytes are present.
        this.padding = String.fromCodePoint(padding);
        // The following paddings are used when only 4 of 5 input bytes are present.
        // This should sort between padding and emojis[0]
        this.padding40 = String.fromCodePoint(padding40);
        // This should sort between emojis[255] and emojis[256]
        this.padding41 = String.fromCodePoint(padding41);
        // This should sort between emojis[511] and emojis[512]
        this.padding42 = String.fromCodePoint(padding42);
        // This should sort between emojis[767] and emojis[768]
        this.padding43 = String.fromCodePoint(padding43);
        this.emojis = emojis.reduce((Map, hexUnicode, i) => Map.set(String.fromCodePoint(Number(hexUnicode)), i), new Map());
        this.emojis.set(this.padding, -1);
        this.revEmojis = emojis.reduce((Map, hexUnicode, i) => Map.set(i, String.fromCodePoint(Number(hexUnicode))), new Map());
        this.revEmojis.set(-1, this.padding);
    }
    getEmoji(id) {
        const emoji = this.revEmojis.get(id);
        if (undefined === emoji) {
            throw Error(`Invalid id provided: ${id}`);
        }
        return emoji;
    }
    getId(rune) {
        const id = this.emojis.get(rune);
        if (undefined === id) {
            throw Error(`Invalid rune provided: ${rune}`);
        }
        return id;
    }
    getPadding() {
        return this.padding;
    }
    getPadding40() {
        return this.padding40;
    }
    getPadding41() {
        return this.padding41;
    }
    getPadding42() {
        return this.padding42;
    }
    getPadding43() {
        return this.padding43;
    }
}
exports.default = Mapping;
